src/main/scala/is/hail/methods/LDMatrix.scala

```diff
@@ -0,0 +1,49 @@
+package is.hail.methods
+
+import is.hail.utils._
+import is.hail.stats.ToNormalizedIndexedRowMatrix
+import is.hail.variant.{Variant, VariantDataset}
+import is.hail.utils.richUtils.RichIndexedRowMatrix._
+import org.apache.spark.mllib.linalg.distributed.{IndexedRow, IndexedRowMatrix}
+
+
+object LDMatrix {
+
+  /**
+    * Computes the LD matrix for the given VDS.
+    * @param vds The VDS to compute LD matrix of.
+    * @return An LDMatrix.
+    */
+  def apply(vds : VariantDataset): LDMatrix = {
+    val nSamples = vds.nSamples
+    val originalVariants = vds.variants.collect().sorted
+
+    val normalizedIRM = ToNormalizedIndexedRowMatrix(vds)
+    normalizedIRM.rows.persist()
+    val variantIndices = normalizedIRM.rows.map{case IndexedRow(idx, _) => idx.toInt}.collect().sorted
+    val variantsUsed = variantIndices.map(idx => originalVariants(idx))
+
+    info(s"Computing LDMatrix with ${variantsUsed.length} variants and $nSamples samples")
+
+    val normalizedFilteredRows = normalizedIRM.rows.zipWithIndex()
+      .map {case (IndexedRow(_, data), idx) => IndexedRow(idx, data)}
+
+    val normalizedBlockMatrix = new IndexedRowMatrix(normalizedFilteredRows).toBlockMatrixDense()
+
+    val indexedRowMatrix = (normalizedBlockMatrix multiply  normalizedBlockMatrix.transpose)
+      .toIndexedRowMatrix()
+
+    val scaledIndexedRowMatrix = new IndexedRowMatrix(indexedRowMatrix.rows.
+      map{case IndexedRow(idx, vals) => IndexedRow(idx, vals.map(d => d / nSamples))})
```

Comment:
Does this make another pass on the data, so that it's worth instead including mult by 1 / sqrt(n) in the ToNormalizedIndexedRowMatrix step, controlled by Boolean flag?  I believe we do 1 / sqrt(m) in RRM or GRM case right (compare also with HWENormalized)

either way, probably better to compute 1 / n first and then multiply by that a billion times.